### PR TITLE
Upgrade add-layers to ng9

### LINKED ISSUE #947: Drop widgets module in favor of UiExtensions module

### LINKED ISSUE #950: Rewrite get-capabilities common module to ng9
This one would also deserve some check for code duplication between get-capabalities, compositions, addLayers and datasources.

### PR DESCRIPTION
**Pull request by jmacura**
also closes #947 and closes #950

### CODE REVIEW

**Comment by jmacura** [comment]
There has been a huge progress in the last days, yet there are still multiple errors underneath:
- unit tests fail because of invalid integration
- ! hs-add-layers-url does not communicate with the rest of the components as it should (I don't know how to manage that) --> the components are currently not in any useful state

I wouldn't mind if some of you join me and help fixing these. I certainly will not have time for it tomorrow.

**Comment by FilipLeitner** [comment]
> There has been a huge progress in the last days, yet there are still multiple errors underneath:
> 
> * unit tests fail because of invalid integration
> * ! hs-add-layers-url does not communicate with the rest of the components as it should (I don't know how to manage that) --> the components are currently not in any useful state
> 
> I wouldn't mind if some of you interfere and help me fixing these. I certainly will not have time for it tomorrow.

Ill take a look (Y)

File: components/add-layers/common/add-layers-url.directive.html [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
         <div class="input-group-append">
 
-            <button type="button" class="btn btn-secondary" [hidden]="!url.valid" (click)="connect()"><i
+            <button type="button" class="btn btn-secondary" *ngIf="uri.valid" (click)="connect()"><i
                     class="icon-link"></i></button>
-            <button type="button" class="btn btn-danger" [hidden]="!url.error.url"><i
-                    class="icon-alertalt"></i></button>
-            <hs-history-list [clicked]="historySelected" [what]="type"></hs-history-list>
+
+            <!-- Not sure what's this next button supposed to be/do -->
```
Comment (jmacura) [inline components/add-layers/common/add-layers-url.directive.html:11]: I guess it was supposed to warn you when you mistype a URL, which is then invalid...

**Comment by raitisbe** [comment]
Will need to replicate https://github.com/hslayers/hslayers-ng/tree/discard-layer-metadata brach commits into this branches typescript files.

**Comment by jmacura** [comment]
> 
> 
> Will need to replicate https://github.com/hslayers/hslayers-ng/tree/discard-layer-metadata brach commits into this branches typescript files.

Thanks for pointing that out. I will re-apply the changes in components/add-layers/vector/VectorLayerDescriptor.js into components/add-layers/vector/VectorLayerDescriptor.ts during the interactive rebase or will re-add them afterwards.

**Comment by jmacura** [comment]
The last commit shall be fixing the add-layers-shp, but I haven't tested it alive though. From localhost it ends with 'Access denied' now, which see as a desired behaviour.

So finally it is ready to review and merge.

**Review by raitisbe — APPROVED** [review]

**Pull request merged**